test/functional/tests/io_class/test_io_class_stats_file_size_core.py

```diff
@@ -0,0 +1,234 @@
+            sync()
+
+    with TestRun.step("In default IO class configuration file set Eviction priority=1 "
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(caches, io_class_sizes)
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = prepare_fio_sizes(io_class_sizes)
+        size_min = Size(512, Unit.Byte)
+
+        for io_class, size in zip(io_classes[:-1], sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device and check that "
+                                  "only for tested IO class statistics increases."):
+                    for core in cores:
+                        core.reset_counters()
+                        fio = fio_params(core, size_min, size)
+                        result = fio.run()
+                        sync()
+                        drop_caches()
+                        issued_reqs_no = \
+                            result[0].write_requests_number() + result[0].read_requests_number()
+                        check_statistics(caches[core.cache_id-1], core, io_classes[:-1],
+                                         io_class, issued_reqs_no)
+                        fs_utils.remove(f"{core.mount_point}/*", force=True, recursive=True)
+                    size_min = size + Size(1, Unit.Byte)
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_stats_file_size_core_direct():
+    """
+    title: Open CAS statistics values for file size - per core.
+    description: Check Open CAS ability to assign correct IO class to file size.
+    pass_criteria:
+        - after FIO with direct statistics increases only for direct IO class
+    """
+
+    with TestRun.step("Prepare devices."):
+        cache_devices, core_devices = prepare_devices()
+
+    with TestRun.step("Start caches (one for each supported cache mode) and add core devices."):
+        caches, cores = prepare_caches_and_cores(cache_devices, core_devices)
+
+    with TestRun.step("In default IO class configuration file set Eviction priority=1 "
+                      "and load it for all caches."):
+        io_classes = prepare_io_classes(caches, io_class_sizes)
+
+    with TestRun.step("Run fio for all devices checking IO class statistics."):
+        sizes = prepare_fio_sizes(io_class_sizes)
+        size_min = Size(512, Unit.Byte)
+
+        io_class_direct = [io_class for io_class in io_classes if "direct" in io_class.rule][0]
+
+        for io_class, size in zip(io_classes[:-1], sizes):
+            with TestRun.step(f"Testing IO class {io_class.id}."):
+                with TestRun.step("Run fio for each device with direct=1 and check that only"
+                                  "for direct IO class statistics increases."):
+                    for core in cores:
```

Comment:
`for cache, core in zip(caches, cores):` here would allow you to substitute `caches[core.cache_id-1]` in line 115 with simple `cache`